harness/determined/cli/workspace.py

```diff
@@ -111,39 +112,34 @@ def list_workspaces(args: Namespace) -> None:
 @authentication.required
 def list_workspace_projects(args: Namespace) -> None:
     sess = cli.setup_session(args)
-    w = workspace_by_name(sess, args.workspace_name)
-    orderArg = bindings.v1OrderBy[args.order_by.upper()]
-    sortArg = bindings.v1GetWorkspaceProjectsRequestSortBy[args.sort_by.upper()]
-    internal_offset = args.offset if ("offset" in args and args.offset) else 0
-    limit = args.limit if "limit" in args else 200
-    all_projects: List[bindings.v1Project] = []
-
-    while True:
-        projects = bindings.get_GetWorkspaceProjects(
-            sess,
-            id=w.id,
-            limit=limit,
-            offset=internal_offset,
-            orderBy=orderArg,
-            sortBy=sortArg,
-        ).projects
-        all_projects += projects
-        internal_offset += len(projects)
-        if ("offset" in args and args.offset) or len(projects) < limit:
-            break
+    all_projects = workspace.Workspace(
+        session=sess, workspace_name=args.workspace_name
+    ).list_projects()
+
+    sort_key = args.sort_by.lower()
+    sort_order = args.order_by.lower()
+    offset = args.offset or 0
+    limit = args.limit or 200
+
+    # TODO: Remove typechecking suppression when mypy is upgraded to 1.4.0
+    all_projects.sort(
+        key=lambda p: getattr(p, sort_key),  # type: ignore
+        reverse=sort_order == "desc",
+    )
+    projects = all_projects[offset : offset + limit]
```

Comment:
hmm... this seems weird to me. i sort of doubt that offset/limit is even needed here, but if it is, should it go in list_projects/REST API?

Reply:
I believe that usage in both CLI and SDK of `offset` is going to be incredibly rare, and that `limit` only somewhat more common.

Keeping methods simple until product or a customer asks for more is one of the parts of the manifesto, and I'd prefer to not implement it there. It's so easy for a user to do, themselves, if they really need it (minus the extra data/time over the wire from filtering client side).

I wouldn't have implemented it here, either, except it was already done and it just didn't feel like the time to remove functionality. Also, I don't mind as much having the flags in the CLI as in the SDK -- it's _marginally_ harder for a user to add themselves, and if you just want to glance at a couple lines it's got _some_ functionality.

I mean, I still wouldn't have offered it if it didn't already exist. I just don't have the same idea of purity as if it was SDK.